### DISCUSSION TITLE
AsyncValue.guard for StreamNotifierProvider

### DISCUSSION THREAD

**Discussion by MkamaMakoko** [rrousselGit/riverpod#2395] [Q&A]
For `AsyncNotifier`, we use
```
state = await AsyncValue.guard(()async{
await modifyData();
return await fetchData();
});
```
This help us to automatically handle exceptions.

But for `StreamNotifierProvider`, we don't need to return anything in `AsyncValue.guard` because the stream will update when new data arrives.

So, what to use when we do want to run a function that might throw exception inside `StreamNotifierProvider`?

**Comment by joakimunge** [comment]
`guard` is nothing more than a shorthand for `try/catch`, so I presume you could just use `try/catch` inside the `StreamNotifierProfider`, but forgive me if I am not understanding the question.

**Reply by MkamaMakoko** [reply]
You are right, I just thought there was a special function for it.
